docs/basics/101-2-populate.rst

```diff
@@ -0,0 +1,213 @@
+with the ``-m`` option:
+
+.. runrecord:: _examples/DL-101-25
+   :language: console
+   :workdir: dl-101
+   :realcommand: cd DataLad-101 && datalad save -m "add books on Python and Unix to read later"
+
+   $ datalad save -m "add books on Python and Unix to read later"
+
+.. container:: toggle
+
+   .. container:: header
+
+      **Addition: "Oh no! I forgot the -m option!"**
+
+   If you forget to specify a commit message with ``-m``, DataLad will write
+   ``[DATALAD] Recorded changes`` as a commit message into your history.
+   This is not particularly informative.
+   You can change the *last* commit message with the Git command
+   ``git commit --amend``. This will open up your default editor
+   and you can edit
+   the commit message. Careful - the default editor might be :term:`vim`!
+
+Lets see how this shows up in the history of the dataset with ``git log``.
+In order to get a bit more details, we add the ``-p`` flag (leave the git log
+by typing ``q``, navigate with up and down arrow keys):
+
+.. runrecord:: _examples/DL-101-26
+   :language: console
+   :workdir: dl-101
+   :lines: 1-20
+   :emphasize-lines: 3-4, 6, 8, 12, 16, 20
+   :realcommand: cd DataLad-101 && git log -p
+
+   $ git log -p
+
+Now this might look a bit cryptic (and honestly, tig [#f1]_ makes it look prettier).
+But this tells us the date and time in which a particular author added two PDFs to
+the directory ``books/``, and thanks to that commit message we have a nice human-
+readable summary of that action.
+
+.. container:: toggle
+
+    .. container:: header
+
+       **Addition: DOs and DON'Ts for commit messages**
+
+    **DOs**
+
+    - Write a *title line* with 72 characters or less (as we did so far)
+
+    - it should be in imperative voice, e.g. "Add notes from lecture 2"
+
+    - Often, a title line is not enough to express your changes and reasoning behind it. In this case, add a body to your commit message by hitting enter twice (before closing the quotation marks), and continue writing a brief summary of the changes after a blank line. This summary should explain "what" has been done and "why", but not "how". Close the quotation marks, and hit enter to save the change with your message.
+
+    - here you can find more guidelines: https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53
+
+    **DON'Ts**
+
+    - passive voice is hard to read afterwards
+
+    - extensive formatting (hashes, asterisks, quotes, ...) will most likely make your shell complain
+
+    - it should be obvious: do not say nasty things about other people
+
+.. gitusernote::
+
+   Just as in Git, new files are not tracked from their creation on, but only when
+   explicitly given to Git (in Git terms with an initial ``git add``). But different
+   from the common Git workflow, DataLad skips the staging area. A ``datalad save``
+   combines a ``git add`` and a ``git commit``, and therefore, the commit message
+   is specified with ``datalad save``.
```

Comment:
FTR: The reason for not using "staging" are limitations of some git-annex modes. Moreover, I personally think that it is a needless complication that is better approached by extensive commiting, and post-hoc history tuning (if desired).